test/roundtripValidatorTests.ts

```diff
@@ -0,0 +1,191 @@
+        ],
+        swaggerPaths: filePaths,
+        loadValidatorInInitialize: true,
+      };
+      const validator = new LiveValidator(options);
+      await validator.initialize();
+
+      assert.equal(validator.operationLoader, undefined);
+    });
+
+    it("OperationLoader should be completely initialized", async () => {
+      console.log("OperationLoader should be completely initialized");
+      const swaggerPattern = "specification/compute/resource-manager/Microsoft.Compute/stable/2021-11-01/runCommands.json";
+      const glob = require("glob");
+      const filePaths: string[] = glob.sync(swaggerPattern, {
+        ignore: DefaultConfig.ExcludedExamplesAndCommonFiles,
+        nodir: true,
+      });
+      const options = {
+        directory: "./test/liveValidation/swaggers/specification",
+        swaggerPathsPattern: [
+          "compute/resource-manager/Microsoft.Compute/stable/2021-11-01/runCommands.json"
+        ],
+        swaggerPaths: filePaths,
+        enableRoundTripValidator: true,
+        enableRoundTripLazyBuild: false
+      };
+      const validator = new LiveValidator(options);
+      await validator.initialize();
+
+      const readOnlys = validator.operationLoader.getAttrs("microsoft.compute", "2021-11-01", "VirtualMachineRunCommands_List", "readOnly");
+      expect(readOnlys).toMatchSnapshot();
+    });
+
+    it("OperationLoader should be initialized only with spec", async () => {
+      console.log("OperationLoader should be initialized only with spec");
+      const swaggerPattern = "specification/compute/resource-manager/Microsoft.Compute/stable/2021-11-01/*.json";
+      const glob = require("glob");
+      const filePaths: string[] = glob.sync(swaggerPattern, {
+        ignore: DefaultConfig.ExcludedExamplesAndCommonFiles,
+        nodir: true,
+      });
+      const options = {
+        directory: "./test/liveValidation/swaggers/specification",
+        swaggerPathsPattern: [
+          "compute/resource-manager/Microsoft.Compute/stable/2021-11-01/*.json"
+        ],
+        swaggerPaths: filePaths,
+        enableRoundTripValidator: true,
+        enableRoundTripLazyBuild: true
+      };
+      const validator = new LiveValidator(options);
+      await validator.initialize();
+
+      let op = validator.operationLoader.cache.get("microsoft.compute")?.get("2021-11-01")?.get("AvailabilitySets_CreateOrUpdate");
+      assert.equal(op, undefined);
+      const readOnlys = validator.operationLoader.getAttrs("microsoft.compute", "2021-11-01", "AvailabilitySets_CreateOrUpdate", "readOnly");
+      expect(readOnlys).toMatchSnapshot();
+      const spec = validator.operationLoader.cache.get("microsoft.compute")?.get("2021-11-01")?.get("spec");
+      assert.notStrictEqual(spec, undefined);
+      op = validator.operationLoader.cache.get("microsoft.compute")?.get("2021-11-01")?.get("AvailabilitySets_CreateOrUpdate");
+      expect(op).toMatchSnapshot();
```

Comment:
The snapshot seems too complicated to check. We should assert the concrete errors, instead of using snapshot.